sdk/src/builder.rs

```diff
@@ -0,0 +1,1105 @@
+                                action = action.set_software_agent(update.to_owned());
+                                actions_mut[index] = action;
+                            }
+                        }
+                    }
+
+                    claim.add_assertion(&actions)
+                }
+                CreativeWork::LABEL => {
+                    let cw: CreativeWork = manifest_assertion.to_assertion()?;
+
+                    claim.add_assertion_with_salt(&cw, &salt)
+                }
+                Exif::LABEL => {
+                    let exif: Exif = manifest_assertion.to_assertion()?;
+                    claim.add_assertion_with_salt(&exif, &salt)
+                }
+                _ => match &manifest_assertion.data {
+                    AssertionData::Json(value) => claim.add_assertion_with_salt(
+                        &User::new(&manifest_assertion.label, &serde_json::to_string(&value)?),
+                        &salt,
+                    ),
+                    AssertionData::Cbor(value) => claim.add_assertion_with_salt(
+                        &UserCbor::new(&manifest_assertion.label, serde_cbor::to_vec(value)?),
+                        &salt,
+                    ),
+                },
+            }?;
+        }
+
+        Ok(claim)
+    }
+
+    // Convert a Manifest into a Store
+    fn to_store(&self) -> Result<Store> {
+        let claim = self.to_claim()?;
+        // commit the claim
+        let mut store = Store::new();
+        let _provenance = store.commit_claim(claim)?;
+        Ok(store)
+    }
+
+    #[cfg(feature = "add_thumbnails")]
+    fn maybe_add_thumbnail<R>(&mut self, format: &str, stream: &mut R) -> Result<&mut Self>
+    where
+        R: Read + Seek + ?Sized,
+    {
+        if self.definition.thumbnail.is_none() {
+            stream.rewind()?;
+            if let Ok((format, image)) =
+                crate::utils::thumbnail::make_thumbnail_from_stream(format, stream)
+            {
+                stream.rewind()?;
+                self.resources
+                    .add(&self.definition.instance_id.clone(), image)?;
+                self.definition.thumbnail = Some(ResourceRef::new(
+                    format,
+                    self.definition.instance_id.clone(),
+                ));
+            }
+        }
+        Ok(self)
+    }
+
+    /// Embed a signed manifest into a stream using a supplied signer.
+    /// # Arguments
+    /// * `format` - The format of the stream
+    /// * `source` - The stream to read from
+    /// * `dest` - The stream to write to
+    /// * `signer` - The signer to use
+    /// # Returns
+    /// * The bytes of c2pa_manifest that was embedded.
+    /// # Errors
+    /// * If the manifest cannot be signed.
+    #[async_generic(async_signature(
+        &mut self,
+        format: &str,
+        source: &mut R,
+        dest: &mut W,
+        signer: &dyn AsyncSigner
+    ))]
+    pub fn sign<R, W>(
```

Comment:
Perhaps this method name should be `sign_and_embed`? The name `sign` by itself doesn't convey the "embed" part of what you wrote in the doc comment above.